physical/yugabytedb-ysql/yugabytedb-ysql_test.go

```diff
@@ -0,0 +1,413 @@
+				// Our test environment is slow enough that we failed this, retry
+				return false
+			}
+			t.Fatalf("should be held")
+		}
+		if val != "baz" {
+			t.Fatalf("bad value: %v", val)
+		}
+	}
+	// The first lock should have lost the leader channel
+	select {
+	case <-time.After(longRenewInterval * 2):
+		t.Fatalf("original lock did not have its leader channel closed.")
+	case <-leaderCh:
+	}
+	return true
+}
+
+// Verify that once Unlock is called, we don't keep trying to renew the original
+// lock.
+func testysqlSQLLockRenewal(t *testing.T, ha physical.HABackend) {
+	// Get the lock
+	origLock, err := ha.LockWith("pgrenewal", "bar")
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+
+	// customize the renewal and watch intervals
+	lock := origLock.(*YSQLLock)
+	// lock.renewInterval = time.Second * 1
+
+	// Attempt to lock
+	leaderCh, err := lock.Lock(nil)
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+	if leaderCh == nil {
+		t.Fatalf("failed to get leader ch")
+	}
+
+	// Check the value
+	held, val, err := lock.Value()
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+	if !held {
+		t.Fatalf("should be held")
+	}
+	if val != "bar" {
+		t.Fatalf("bad value: %v", val)
+	}
+
+	// Release the lock, which will delete the stored item
+	if err := lock.Unlock(); err != nil {
+		t.Fatalf("err: %v", err)
+	}
+
+	// Wait longer than the renewal time
+	time.Sleep(1500 * time.Millisecond)
+
+	// Attempt to lock with new lock
+	newLock, err := ha.LockWith("pgrenewal", "baz")
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+
+	stopCh := make(chan struct{})
+	timeout := time.Duration(lock.ttlSeconds)*time.Second + lock.retryInterval + time.Second
+
+	var leaderCh2 <-chan struct{}
+	newlockch := make(chan struct{})
+	go func() {
+		leaderCh2, err = newLock.Lock(stopCh)
+		close(newlockch)
+	}()
+
+	// Cancel attempt after lock ttl + 1s so as not to block unit tests forever
+	select {
+	case <-time.After(timeout):
+		t.Logf("giving up on lock attempt after %v", timeout)
+		close(stopCh)
+	case <-newlockch:
+		// pass through
+	}
+
+	// Attempt to lock should work
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+	if leaderCh2 == nil {
+		t.Fatalf("should get leader ch")
+	}
+
+	// Check the value
+	held, val, err = newLock.Value()
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+	if !held {
+		t.Fatalf("should be held")
+	}
+	if val != "baz" {
+		t.Fatalf("bad value: %v", val)
+	}
+
+	// Cleanup
+	newLock.Unlock()
+}
+
+func setupDatabaseObjects(t *testing.T, logger log.Logger, yb *YSQLBackend) {
+	var err error
+	// Setup tables and indexes if not exists.
+	createTableSQL := fmt.Sprintf(
+		"  CREATE TABLE IF NOT EXISTS %v ( "+
+			"  parent_path TEXT COLLATE \"C\" NOT NULL, "+
+			"  path        TEXT COLLATE \"C\", "+
+			"  key         TEXT COLLATE \"C\", "+
+			"  value       BYTEA, "+
+			"  CONSTRAINT pkey PRIMARY KEY (path, key) "+
+			" ); ", yb.table)
```

Comment:
You can use raw strings (using backticks) that allows including new lines as part of a string. Also no escaping will be required for the double quotes.